drivers/regex/mlx5/mlx5.c

```diff
@@ -255,12 +312,16 @@ mlx5_regex_pci_probe(struct rte_pci_driver *pci_drv __rte_unused,
 		rte_errno = ENOTSUP;
 		return -rte_errno;
 	} else {
+#ifndef REGEX_MLX5_NO_REAL_HW
 		if (!attr.regex || attr.regexp_num_of_engines == 0) {
 			DRV_LOG(ERR, "Not enough capabilities to support regex,"
 				" maybe old FW/OFED version?");
 			rte_errno = ENOTSUP;
 			return -rte_errno;
 		}
+#else
+		attr.regexp_num_of_engines = 2;
```

Comment:
My API will fill relevant attributes when REGEX_MLX5_NO_REAL_HW defined
No need to do this here